### PR TITLE
refactor: remove extended collection

### PR DESCRIPTION
**Pull request by 1Computer1**
**Please describe the changes this PR makes and why it should be merged:**

Depends on https://github.com/discordjs/collection/pull/44. Removes usage of `Util.flatten` on entries that do not have `toJSON`.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import {
   TimestampStylesString,
   underscore,
 } from '@discordjs/builders';
-import BaseCollection from '@discordjs/collection';
+import Collection from '@discordjs/collection';
```
Comment (kyranet) [inline typings/index.d.ts:16]: For consistency with the JS-side:
```suggestion
import { Collection } from '@discordjs/collection';
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Good riddance, just one thing, LGTM otherwise

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by Shivakumar14** [comment]
Unsubscribe

**Comment by Techbot121** [comment]
https://github.com/discordjs/discord.js/blob/master/src/util/Util.js#L41 I assume this slipped by?

**Pull request merged**